ethereum/core/src/main/java/org/hyperledger/besu/ethereum/storage/keyvalue/KeyValueSegmentIdentifier.java

```diff
@@ -46,32 +49,37 @@ public enum KeyValueSegmentIdentifier implements SegmentIdentifier {
   CHAIN_PRUNER_STATE(new byte[] {18});
 
   private final byte[] id;
-  private final int[] versionList;
+  private final EnumSet<DataStorageFormat> formats;
   private final boolean containsStaticData;
   private final boolean eligibleToHighSpecFlag;
   private final boolean staticDataGarbageCollectionEnabled;
 
   KeyValueSegmentIdentifier(final byte[] id) {
-    this(id, new int[] {0, 1, 2});
+    this(id, EnumSet.allOf(DataStorageFormat.class));
   }
 
   KeyValueSegmentIdentifier(
       final byte[] id, final boolean containsStaticData, final boolean eligibleToHighSpecFlag) {
-    this(id, new int[] {0, 1, 2}, containsStaticData, eligibleToHighSpecFlag, false);
+    this(
+        id,
+        EnumSet.allOf(DataStorageFormat.class),
+        containsStaticData,
+        eligibleToHighSpecFlag,
+        false);
   }
 
-  KeyValueSegmentIdentifier(final byte[] id, final int[] versionList) {
```

Comment:
That's a good question, in theory the column families (identifiers) are a concept that should not be in the plugin API, since a storage plugin should be free to organized the data as it wants, we can decide to review also this part later.

For the moment the specific version of the format is hidden here, since it is an internal property of each plugin, but that should not limit too much, since if you want to change the column families a new version of Besu is released and the new version of the RocksDB embedded plugin can managed the upgrade/downgrade.

Reply:
Good point we do the versioning at the application level. I actually prefer that than we have right now.